src/registrar/models/models.py

```diff
@@ -0,0 +1,422 @@
+    entity = models.ForeignKey(GovEntity, on_delete=models.CASCADE)
+    authorizing_official = models.ForeignKey(AuthorizingOfficial, on_delete=models.CASCADE)
+
+
+class Rejection(models.Model):
+    reason = models.CharField(max_length=2)
+
+
+class Application(TimeStampedModel):
+    # user who is making the request
+    user = models.ForeignKey(User, null=True, on_delete=models.SET_NULL)
+    # the person who says this user is allowed to request this domain
+    authorizing_official = models.ForeignKey(AuthorizingOfficial, on_delete=models.CASCADE)
+    # the government who CISA says is allowed to request a .gov domain
+    authorizing_entity = models.ForeignKey(GovEntity, on_delete=models.CASCADE)
+    # the subdivision or department this user belongs to
+    organization = models.ForeignKey(Organization, on_delete=models.PROTECT)
+    domain_name = models.CharField(max_length=2)
+    is_exception_requested = models.BooleanField()
+    # if the user wants an ineligible domain name, they may give a reason for their request
+    exception_rationale = models.CharField(max_length=2)
+    is_expedited_requested = models.BooleanField()
+    # if the user wants a fast response, they may give a reason for their request
+    expedited_rationale = models.CharField(max_length=2)
+    hsts_acknowledged = models.BooleanField()
+    # reason for requesting the domain
+    purpose = models.CharField(max_length=2)
+    # Choices ??? -- also, TODO: use a state machine to model the business logic
+    status = models.CharField(max_length=2)
+    # choices ???
+    rejection_codes = models.ManyToManyField(Rejection)
+    assignee = models.ForeignKey(User, null=True, on_delete=models.SET_NULL, related_name="assigned")
+
+
+class ApplicationStatusHistory(TimeStampedModel):
+    application = models.ForeignKey(Application, on_delete=models.CASCADE)
+    prior_status = models.CharField(max_length=2)
+    status = models.CharField(max_length=2)
+    change_note = models.CharField(max_length=2)
+
+
+class Domain(TimeStampedModel, CommonEPPModel):
+    # fully qualified name (unique)
+    name  = models.CharField(max_length=2)
+    # identifier assigned to the domain object when created
+    roid  = models.CharField(max_length=2) # NS1_EXAMPLE1-REP
+    # the date and time of the domain object's registration expiration
+    ex_date = models.CharField(max_length=2)
+    # can have 1..* statuses
+    status = models.ManyToManyField(DomainStatus)
+    organization = models.ForeignKey(Organization, on_delete=models.PROTECT)
+
+
+class Host(TimeStampedModel, CommonEPPModel):
+    # fully qualified name
+    name  = models.CharField(max_length=2)
+    # identifier assigned to the host object when created
+    roid  = models.CharField(max_length=2) # NS1_EXAMPLE1-REP
+    # can have 1..* statuses
+    status = models.ManyToManyField(HostStatus)
+    # domain this host belongs to
+    addr = models.ForeignKey(Domain, on_delete=models.PROTECT, related_name="host")
```

Comment:
Is this another field name that i related to another system? Since we have multiple things called "addresses" here, could we use `domain` for this field? 

The `related_name` should be plural since it will be used as `Domain.hosts`.

Reply:
I'm okay with spelling out abbreviations, but I'd prefer `address` to `domain` because I want to clearly preserve the mapping between EPP and our database. (Spoken as the person who has had to debug too many legacy systems with "name it whatever you want in thirty different places" 😠 )